src/multiparser.js

```diff
@@ -25,64 +26,57 @@ function fromBabylonFlowOrTypeScript(path) {
   const node = path.getValue();
 
   switch (node.type) {
-    case "TemplateElement": {
+    case "TemplateLiteral": {
       const parent = path.getParentNode();
       const parentParent = path.getParentNode(1);
-      const parentParentParent = path.getParentNode(2);
 
       /*
        * styled-jsx:
        * ```jsx
        * <style jsx>{`div{color:red}`}</style>
        * ```
        */
-      if (
-        parentParentParent &&
-        parent.quasis &&
-        parent.quasis.length === 1 &&
-        parentParent.type === "JSXExpressionContainer" &&
-        parentParentParent.type === "JSXElement" &&
-        parentParentParent.openingElement.name.name === "style" &&
-        parentParentParent.openingElement.attributes.some(
+      const isStyledJsx =
+        parentParent &&
+        node.quasis &&
+        parent.type === "JSXExpressionContainer" &&
+        parentParent.type === "JSXElement" &&
+        parentParent.openingElement.name.name === "style" &&
+        parentParent.openingElement.attributes.some(
           attribute => attribute.name.name === "jsx"
-        )
-      ) {
-        return {
-          parser: "postcss",
-          wrap: doc =>
-            concat([
-              indent(concat([softline, stripTrailingHardline(doc)])),
-              softline
-            ]),
-          text: parent.quasis[0].value.raw
-        };
-      }
+        );
 
       /*
        * styled-components:
        * styled.button`color: red`
        * Foo.extend`color: red`
        */
-      if (
-        parentParent &&
-        parentParent.type === "TaggedTemplateExpression" &&
-        parent.quasis.length === 1 &&
-        parentParent.tag.type === "MemberExpression" &&
-        (parentParent.tag.object.name === "styled" ||
-          (/^[A-Z]/.test(parentParent.tag.object.name) &&
-            parentParent.tag.property.name === "extend"))
-      ) {
+      const isStyledComponents =
+        parent &&
+        parent.type === "TaggedTemplateExpression" &&
+        parent.tag.type === "MemberExpression" &&
+        (parent.tag.object.name === "styled" ||
+          (/^[A-Z]/.test(parent.tag.object.name) &&
+            parent.tag.property.name === "extend"));
+
+      if (isStyledJsx || isStyledComponents) {
```

Comment:
nit: `isStyledComponents` and `isStyledJsx` could be functions that get the current `path` as input.

Then you can make an helper function `isCss`:

```js
export const isCss = path => [isStyledJsx, isStyledCompoents].some(test => test(path))
```

This way you can add `test`s for other libraries easily.